lib/dry/logic/build/operation.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+module Dry
+  module Logic
+    module Build
+      class Operation < Base
+        INFLECTOR = Dry::Inflector.new.freeze
+
+        def method_missing(method, *args, **kwargs, &block)
+          super unless respond_to_missing?(method)
+          super unless Kernel.block_given?
+
+          to_operation(method).new(*to_predicate(&block), *args, **kwargs)
+        end
+
+        def respond_to_missing?(method, *)
+          !to_operation(method).nil?
```

Comment:
`to_operation` is called by both `method_missing` and `respond_to_missing?`. Memoize it.

Reply:
Fixed in 424a82d.